default/python/AI/ColonisationAI.py

```diff
@@ -983,46 +999,46 @@ def evaluate_planet(planet_id, mission_type, spec_name, detail=None):
+                        else:
+                            asteroid_factor = 0.0
                         ast_shipyard_name = p2.name
-                        if p2.owner != -1:
-                            got_owned_asteroids = True
                     if p2.size == fo.planetSize.gasGiant:
-                        local_gg = True
-                        if p2.owner != -1:
-                            got_owned_gg = True
-        if got_asteroids:
+                        if p2.owner==empire.empireID:
+                            gg_factor = 1.0
+                        elif p2.unowned:
+                            gg_factor = 0.5
+                        elif pid in prospective_invasion_targets:
+                            gg_factor = character.secondary_valuation_factor_for_invasion_targets()
+                        else:
+                            gg_factor = 0.0
+        if asteroid_factor > 0.0:
             if tech_is_complete("PRO_MICROGRAV_MAN") or "PRO_MICROGRAV_MAN" in empire_research_list[:10]:
-                if got_owned_asteroids:  # can be quickly captured
-                    flat_industry += 5  # will go into detailed industry projection
-                    detail.append("Asteroid mining ~ %.1f" % (5 * discount_multiplier))
-                else:  # uncertain when can be outposted
-                    asteroid_bonus = 2.5 * discount_multiplier  # give partial value
-                    detail.append("Asteroid mining %.1f" % (5 * discount_multiplier))
+                flat_industry += 5 * asteroid_factor  # will go into detailed industry projection
+                detail.append("Asteroid mining ~ %.1f" % (5 * asteroid_factor * discount_multiplier))
             if tech_is_complete("SHP_ASTEROID_HULLS") or "SHP_ASTEROID_HULLS" in empire_research_list[:11]:
                 if species and species.canProduceShips:
-                    asteroid_bonus += 30 * discount_multiplier * pilot_val
+                    asteroid_bonus = 30 * discount_multiplier * pilot_val
                     detail.append(
                         "Asteroid ShipBuilding from %s %.1f" % (
-                            ast_shipyard_name, discount_multiplier * 20 * pilot_val))
-        if local_gg:
+                            ast_shipyard_name, discount_multiplier * 30 * pilot_val))
+        if gg_factor > 0.0:
             if tech_is_complete("PRO_ORBITAL_GEN") or "PRO_ORBITAL_GEN" in empire_research_list[:5]:
-                if got_owned_gg:
-                    flat_industry += per_ggg  # will go into detailed industry projection
-                    detail.append("GGG ~ %.1f" % (per_ggg * discount_multiplier))
-                else:
-                    gas_giant_bonus = 0.5 * per_ggg * discount_multiplier
-                    detail.append("GGG %.1f" % (0.5 * per_ggg * discount_multiplier))
+                flat_industry += per_ggg * gg_factor  # will go into detailed industry projection
```

Comment:
See comment above re ```flat_industry```

Reply:
For the same reasons as above, I think it better to have this go into ```flat_industry```; I think the ```gas_giant_bonus``` is now unused/unneeded and I should just take it out entirely.